### ISSUE TITLE
Ability to use elements within Select values

### ISSUE THREAD

**Issue by ckknight** [lingui/js-lingui#1288]
TL;DR: `<Select>` should either support non-string values or be extremely explicit in the documentation about why they can't/shouldn't be used.

I was attempting to use `<Select>` earlier and for one of the values, I realized I needed to use a nested element (and thus not just rely on a string).

I had attempted to use the following syntax:

```tsx
<Select
  value="happy"
  happy={
    <Trans>
      Hooray! <Icon />
    </Trans>
  }
  sad={
    <>
      Oh no! <Icon />
    </>
  }
  other="Dunno"
/>
```

This has three immediate issues that I can see
1. TypeScript expects _only_ `string` properties here
2. `lingui extract` complains with a message like
    ``` 
    Missing message ID, skipping.
    <Trans __self={this} __source={{
      fileName: _jsxFileName,
      lineNumber: 115,
      columnNumber: 17
    }}>
    ```
3. This error occurs at runtime: Duplicate __self prop found. You are most likely using the deprecated transform-react-jsx-self Babel plugin. Both __source and __self are automatically set when using the automatic runtime. Please remove transform-react-jsx-source and transform-react-jsx-self from your Babel config.

So it seems that `Trans`-inside-`Select` isn't great, even though the `.po` files do look _mostly_ accurate. (`msgstr "{0, select, happy { Hooray! <0/>} sad {{1}} other {Dunno}}"`)

Upon trying without the nested syntax,

```tsx
<Select
  value="happy"
  happy={
    <>
      Hooray! <Icon />
    </>
  }
  sad={
    <>
      Oh no! <Icon />
    </>
  }
  other="Dunno"
/>
```

This does appear to work at runtime, but "Hooray!" and "Oh no!" become untranslatable. The TypeScript issue is still a problem.

The following _does_ seem to be the closest I've come to actually working:
```tsx
<Trans>
  <Select
    value="happy"
    happy={
      <Trans>
        Hooray! <Icon />
      </Trans>
    }
    sad={
      <Trans>
        Oh no! <Icon />
      </Trans>
    }
    other="Dunno"
  />
</Trans>
```

This appears to do what I intend to do at runtime.

In the po file, the entry looks like
```po
msgstr "{0, select, happy { Hooray! <0/>} sad { Oh no! <1/>} other {Dunno}}"
```

Given the extra spaces, it appears that if one puts a parenthesis or some character at the start of the translation, before the `<Select>`, an unexpected whitespace character slips in.

**Comment by thekip** [comment]
@andrii-bodnar i will take this

**Comment by thekip** [comment]
Well, this related to this one https://github.com/lingui/js-lingui/issues/1324

Exact cases must be with underscore, documentation is wrong 😬

```ts
<Select
          value="happy"
          _happy={
            <Trans>
              Hooray! <Icon />
            </Trans>
          }
          _sad={
            <Trans>
              Oh no! <Icon />
            </Trans>
          }
          other="Dunno"
        />
```

Builds perfectly into this: 

```ts
import { Trans } from "@lingui/react";
<Trans
  id={"{0, select, happy { Hooray! <0/>} sad { Oh no! <1/>} other {Dunno}}"}
  values={{
    0: "happy",
  }}
  components={{
    0: <Icon />,
    1: <Icon />,
  }}
/>;
```

But still TS typings is wrong, they accept only string, but actually they should accept ReactNode.

**Comment by Martin005** [comment]
Documentation fixed by #1421 and typings fixed by #1340. This will be part of the v4.
